src/esp/gfx/CubeMap.cpp

```diff
@@ -0,0 +1,399 @@
+#endif
+}
+
+void CubeMap::renderToTexture(CubeMapCamera& camera,
+                              scene::SceneGraph& sceneGraph,
+                              RenderCamera::Flags flags) {
+  CORRADE_ASSERT(camera.isInSceneGraph(sceneGraph),
+                 "CubeMap::renderToTexture(): camera is NOT attached to the "
+                 "current scene graph.", );
+  // ==== projection matrix ====
+  // CAREFUL! In this function the projection matrix of the camera is assumed to
+  // be set already outside of this function by the user.
+  // we simply do sanity check here.
+  {
+    Mn::Vector2i vp = camera.viewport();
+    CORRADE_ASSERT(vp.x() == vp.y() && vp.x() == imageSize_,
+                   "CubeMap::renderToTexture(): the image size with in the "
+                   "CubeMapCamera, which is"
+                       << vp.x() << "by" << vp.y() << "compared to"
+                       << imageSize_ << "is not correct.", );
+  }
+
+  // ==== camera matrix ====
+  // In case user set the relative transformation of
+  // the camera node before calling this function, original viewing matrix of
+  // the camera MUST be updated as well.
+  camera.updateOriginalViewingMatrix();
+  frameBuffer_.bind();
+  for (int iFace = 0; iFace < 6; ++iFace) {
+    camera.switchToFace(iFace);
+    prepareToDraw(iFace);
+
+    // TODO:
+    // camera should have flags so that it can do "low quality" rendering,
+    // e.g., no normal maps, no specular lighting, low-poly meshes,
+    // low-quality textures.
+
+    for (auto& it : sceneGraph.getDrawableGroups()) {
+      // TODO: remove || true
+      if (it.second.prepareForDraw(camera) || true) {
+        camera.draw(it.second, flags);
+      }
+    }
+  }  // iFace
+
+  // CAREFUL!!!
+  // switchToFace() will change the local transformation of this camera node!
+  // If you do not do anything, in next rendering cycle, since
+  // camera.updateOriginalViewingMatrix() is called, the original viewing matrix
+  // will be updated by mistake!!! To prevent such mistakes, local
+  // transformation of this camera node must be reset.
+  camera.restoreTransformation();
+
+  // Color texture ONLY, NOT for depth
+  if (flags_ & (Flag::BuildMipMap | Flag::ColorTexture)) {
+    textures_[TextureType::Color]->generateMipmap();
+  }
+}
+
+void CubeMap::loadTexture(TextureType type,
+                          const std::string& imageFilePrefix,
+                          const std::string& imageFileExtension) {
+  auto loadImporter = [&](const std::string& plugin) {
+    Cr::PluginManager::Manager<Mn::Trade::AbstractImporter> manager;
+    Cr::Containers::Pointer<Mn::Trade::AbstractImporter> importer =
+        manager.loadAndInstantiate(plugin);
+    CORRADE_ASSERT(importer,
+                   "CubeMap::loadTexture(): cannot initialize" << plugin, );
+
+    imageImporterManger_.set<Mn::Trade::AbstractImporter>(
+        plugin, importer.release(), Mn::ResourceDataState::Final,
+        Mn::ResourcePolicy::Manual);
```

Comment:
Wait this *can't* work, the `importer` is not allowed to outlive the `manager` (because in case of dynamic plugins the manager is responsible for `dlopen()`/`dlclose()`ing the plugin binary and other reasons). 

If it works and doesn't assert then it's maybe because for static plugins it's *mostly* fine, but is generally not a correct workflow.

Reply:
This might be the reason of the super rare segfault I mentioned to you a while ago.